Enable NeoC# node image builds for macOS

## Is your feature request related to a problem? Please describe.
NeoGo image is building OK on macOS, but NeoC# image is not. First of all, building from official release binaries (with `DF_SHARP=.docker/build/Dockerfile.sharp`) is not going to work since binaries are built for another architecture/OS which results in unexpected sigfaults on program attempt to interact with memory. Secondly, I'd expect that building in-place (with `DF_SHARP=.docker/build/Dockerfile.sharp.sources.from_local_dependencies`) should work, but the process requires additional adjustment:
* Building without any additional modifications results in:
```
qemu-x86_64: Could not open '/lib64/[ld-linux-x86-64.so](http://ld-linux-x86-64.so/).2': No such file or directory
```
* Building with `--platform linux/amd64` results in:
```
/usr/bin/make -f /Users/ekaterinapavlova/Workplace/neo-bench/Makefile -C /Users/ekaterinapavlova/Workplace/neo-bench build.node.sharp
=> Building Sharp Node image registry.nspcc.ru/neo-bench/neo-sharp:bench
#0 building with "desktop-linux" instance using docker driver

#1 [internal] load build definition from Dockerfile.sharp.sources.from_local_dependencies
#1 transferring dockerfile: 4.78kB done
#1 DONE 0.0s

#2 [internal] load metadata for mcr.microsoft.com/dotnet/aspnet:7.0
#2 DONE 1.2s

#3 [internal] load metadata for mcr.microsoft.com/dotnet/sdk:7.0
#3 DONE 1.3s

#4 [internal] load .dockerignore
#4 transferring context: 2B done
#4 DONE 0.0s

#5 [build 1/6] FROM mcr.microsoft.com/dotnet/sdk:7.0@sha256:3ee57ab678f7ecc254dc674d7889fc764698e3b51381c4dd3d1d414ecf0295fb
#5 DONE 0.0s

#6 [final  1/11] FROM mcr.microsoft.com/dotnet/aspnet:7.0@sha256:21f1da4b95d84ad3313920e1deb3d45e416a870b937e9a885cb8eb7ca7bf8ff0
#6 DONE 0.0s

#7 [build 2/6] RUN set -x     && apt-get update     && apt-get install -y         unzip         wget
#7 CACHED

#8 [build 3/6] RUN mkdir /neo-project &&     wget -O /tmp/neo.zip https://github.com/neo-project/neo/archive/v3.6.0.zip &&     unzip -q -d /neo-project/ /tmp/neo.zip &&     mv /neo-project/neo-* /neo-project/neo &&     wget -O /tmp/neo-cli.zip https://github.com/neo-project/neo-node/archive/v3.6.0.zip &&     unzip -q -d /neo-project/ /tmp/neo-cli.zip &&     mv /neo-project/neo-node-* /neo-project/neo-node &&     wget -O /tmp/neo-modules.zip https://github.com/neo-project/neo-modules/archive/v3.6.1.zip &&     unzip -q -d /neo-project/ /tmp/neo-modules.zip &&     mv /neo-project/neo-modules-* /neo-project/neo-modules &&     wget -O /tmp/neo-vm.zip https://github.com/neo-project/neo-vm/archive/v3.6.0.zip &&     unzip -q -d /neo-project/ /tmp/neo-vm.zip &&     mv /neo-project/neo-vm-* /neo-project/neo-vm
#8 CACHED

#9 [build 4/6] RUN sed -i '/PackageReference Include="Neo" Version=/c \    <ProjectReference Include="..\\..\\neo\\src\\Neo\\Neo.csproj" \/>' /neo-project/neo-node/neo-cli/neo-cli.csproj &&     sed -i '/<PackageReference Include="Neo.VM" Version=/c \    <ProjectReference Include="..\\..\\..\\neo-vm\\src\\Neo.VM\\Neo.VM.csproj" \/>' /neo-project/neo/src/Neo/Neo.csproj &&     sed -i '/<PackageReference Include="Neo" Version=/c \    <ProjectReference Include="..\\..\\..\\neo\\src\\Neo\\Neo.csproj" \/>' /neo-project/neo-modules/src/Directory.Build.props
#9 CACHED

#10 [final  2/11] RUN set -x     && apt-get update     && apt-get install -y         libleveldb-dev         curl         jq         libssl-dev         libunwind8         librocksdb-dev         libc6-dev         iproute2     && rm -rf /var/lib/apt/lists/*
#10 CACHED

#11 [internal] load build context
#11 transferring context: 164B done
#11 DONE 0.0s

#12 [build 5/6] RUN dotnet publish -c Release --framework net7.0 -r linux-x64 /neo-project/neo-node/neo-cli -o /neo-cli
#12 1.617 MSBuild version 17.7.4+3ebbd7c49 for .NET
#12 5.950 assertion failed [block != nullptr]: BasicBlock requested for unrecognized address
#12 5.950 (BuilderBase.h:550 block_for_offset)
#12 5.950  Trace/breakpoint trap
#12 ERROR: process "/bin/sh -c dotnet publish -c Release --framework net7.0 -r linux-x64 /neo-project/neo-node/neo-cli -o /neo-cli" did not complete successfully: exit code: 133
------
 > [build 5/6] RUN dotnet publish -c Release --framework net7.0 -r linux-x64 /neo-project/neo-node/neo-cli -o /neo-cli:
1.617 MSBuild version 17.7.4+3ebbd7c49 for .NET
Trace/breakpoint trap
------
Dockerfile.sharp.sources.from_local_dependencies:49
--------------------
  47 |     # Publish neo-cli from source as a self-contained deployment for linux-64 into /neo-cli folder (all dependant .dlls are included).
  48 |     # See https://docs.microsoft.com/ru-ru/dotnet/core/deploying/#publish-self-contained for details.
  49 | >>> RUN dotnet publish -c Release --framework net7.0 -r linux-x64 /neo-project/neo-node/neo-cli -o /neo-cli
  50 |     
  51 |     # Build neo-modules from source into /Plugins folder (only plugin .dll and plugin config are included, if you need other dependant .dlls, see the next step)
--------------------
ERROR: failed to solve: process "/bin/sh -c dotnet publish -c Release --framework net7.0 -r linux-x64 /neo-project/neo-node/neo-cli -o /neo-cli" did not complete successfully: exit code: 133
make: *** [build.node.sharp] Error 1

Process finished with exit code 2
```
https://stackoverflow.com/questions/74735374/can-i-create-a-net6-app-for-macos-with-both-x64-and-arm64-support may be related.

## Describe the solution you'd like
We need to be able to properly build the C# image, but it should be done after 3.7 migration since likely requires `.csproj` files modification.

--- a/.docker/build/Dockerfile.sharp.sources.from_binaries
+++ b/.docker/build/Dockerfile.sharp.sources.from_binaries
@@ -7,7 +7,7 @@ RUN set -x \
         unzip \
         wget
 
-# Publish neo-cli from source as a self-contained deployment for linux-64 into /neo-cli folder (all dependant .dlls are included).
+# Publish neo-cli from source as a self-contained deployment into /neo-cli folder (all dependant .dlls are included).
 # See https://docs.microsoft.com/ru-ru/dotnet/core/deploying/#publish-self-contained for details.
 ENV CLIBRANCH="v3.6.0"
 RUN wget -O /tmp/neo-cli.zip https://github.com/neo-project/neo-node/archive/${CLIBRANCH}.zip && \
@@ -17,7 +17,7 @@ RUN wget -O /tmp/neo-cli.zip https://github.com/neo-project/neo-node/archive/${C
     cp -r tmp/neo-node/*/Neo.ConsoleService /build && \
     cp tmp/neo-node/*/NuGet.Config /build/neo-cli && \
     dotnet restore /build/neo-cli && \
-    dotnet publish -c Release --framework net7.0 -r linux-x64 /build/neo-cli -o /neo-cli && \
+    dotnet publish -c Release --framework net7.0 /build/neo-cli -o /neo-cli && \
     rm -rf /build /tmp/neo-cli.zip /tmp/neo-node
 
 # Build neo-modules from source into /Plugins folder (only plugin .dll and plugin config are included, if you need other dependant .dlls, see the next step)
@@ -42,21 +42,21 @@ RUN dotnet publish -c Release --framework net7.0 /tmp/neo-modules/*/src/RpcServe
     cp /tmp/neo-modules/*/src/RocksDBStore/bin/Release/net7.0/publish/RocksDbSharp.dll /Plugins/ && \
     rm -rf /tmp/neo-modules.zip /tmp/neo-modules
 
-# Publish neo-vm from source as a self-contained deployment for linux-64 into /neo-vm folder (although neo-vm does not have dependant .dlls)
+# Publish neo-vm from source as a self-contained deployment into /neo-vm folder (although neo-vm does not have dependant .dlls)
 ENV NEOVMBRANCH="v3.6.0"
 RUN wget -O /tmp/neo-vm.zip https://github.com/neo-project/neo-vm/archive/${NEOVMBRANCH}.zip && \
     unzip -q -d /tmp/neo-vm/ /tmp/neo-vm.zip && \
     ls -l /tmp/* && \
     dotnet restore /tmp/neo-vm/*/src/Neo.VM/ && \
-    dotnet publish -c Release --framework net7.0 -r linux-x64 /tmp/neo-vm/*/src/Neo.VM/ -o /neo-vm && \
+    dotnet publish -c Release --framework net7.0 /tmp/neo-vm/*/src/Neo.VM/ -o /neo-vm && \
     rm -rf /tmp/neo-vm.zip tmp/neo-vm
 
-# Publish neo from source as a self-contained deployment for linux-64 into /neo folder (all dependant .dlls are included)
+# Publish neo from source as a self-contained deployment into /neo folder (all dependant .dlls are included)
 ENV NEOBRANCH="v3.6.0"
 RUN wget -O /tmp/neo.zip https://github.com/neo-project/neo/archive/${NEOBRANCH}.zip && \
     unzip -q -d /tmp/neo/ /tmp/neo.zip && \
     dotnet restore /tmp/neo/*/src/Neo/ && \
-    dotnet publish -c Release --framework net7.0 -r linux-x64 /tmp/neo/*/src/Neo/ -o /neo && \
+    dotnet publish -c Release --framework net7.0 /tmp/neo/*/src/Neo/ -o /neo && \
     rm -rf /tmp/neo.zip tmp/neo
 
 # All things are published, so build the final image by copying binaries from Build

--- a/.docker/build/Dockerfile.sharp.sources.from_local_dependencies
+++ b/.docker/build/Dockerfile.sharp.sources.from_local_dependencies
@@ -44,9 +44,9 @@ RUN sed -i '/PackageReference Include="Neo" Version=/c \    <ProjectReference In
     sed -i '/<PackageReference Include="Neo.VM" Version=/c \    <ProjectReference Include="..\\..\\..\\neo-vm\\src\\Neo.VM\\Neo.VM.csproj" \/>' /neo-project/neo/src/Neo/Neo.csproj && \
     sed -i '/<PackageReference Include="Neo" Version=/c \    <ProjectReference Include="..\\..\\..\\neo\\src\\Neo\\Neo.csproj" \/>' /neo-project/neo-modules/src/Directory.Build.props
 
-# Publish neo-cli from source as a self-contained deployment for linux-64 into /neo-cli folder (all dependant .dlls are included).
+# Publish neo-cli from source as a self-contained deployment into /neo-cli folder (all dependant .dlls are included).
 # See https://docs.microsoft.com/ru-ru/dotnet/core/deploying/#publish-self-contained for details.
-RUN dotnet publish -c Release --framework net7.0 -r linux-x64 /neo-project/neo-node/neo-cli -o /neo-cli
+RUN dotnet publish -c Release --framework net7.0 /neo-project/neo-node/neo-cli -o /neo-cli
 
 # Build neo-modules from source into /Plugins folder (only plugin .dll and plugin config are included, if you need other dependant .dlls, see the next step)
 RUN mkdir /Plugins && \

--- a/README.md
+++ b/README.md
@@ -264,7 +264,17 @@ The following default configurations are available:
 ## Runner usage (`./runner.sh`)
 
 ```
+   -v, --validators                 Consensus node count.
+                                    Possible values: 1, 4 (default), 7.
+   -n, --nodes                      Consensus node type.
+                                    Possible values: go (default), mixed, sharp.
+   -r, --rpc                        RPC node type. Default is the same as --nodes.
    -h, --help                       Show usage message.
+   -b, --benchmark                  Benchmark type.
+                                    Possible values: NEO (default) or GAS
+       --from                       Number of tx senders (default: 1)
+       --to                         Number of fund receivers (default: 1)
+       --vote                       Whether or not candidates should be voted for before the bench.
    -d                               Benchmark description.
    -m                               Benchmark mode.
                                     Example: -m wrk -m rate
@@ -282,6 +292,11 @@ The following default configurations are available:
    -t                               Request timeout.
                                     Used for RPC requests.
                                     Example: -t 30s
+   -l, --log                        Container logging facility. Default value is none.
+                                    Example: -l journald -l syslog -l json-file
+       --tc                         Arguments to pass to 'tc qdisc netem' inside the container.
+                                    Example: 'delay 100ms'
+
 ```
 
 ## Build options
@@ -410,7 +425,7 @@ To add one more node configuration, provide all necessary information to the `no
 
 Name|Description| Default |Example
 ---|---|---------|---
-NEOBENCH_LOGGER|Container logging facility| `none`  |`none`, `journald`, `syslog`
+NEOBENCH_LOGGER|Container logging facility| `none`  |`none`, `journald`, `syslog`,`json-file`
 NEOBENCH_TC|Parameters passed to the `tc qdisc` (netem discipline) on container startup|         |`delay 100ms`
 NEOBENCH_TYPE|Type of the load| `NEO`   |`NEO`, `GAS`
 NEOBENCH_FROM_COUNT|Number of tx senders| `1`     | `1`
@@ -418,6 +433,8 @@ NEOBENCH_TO_COUNT|Number of fund receivers| `1`     | `1`
 NEOBENCH_VALIDATOR_COUNT|Number of validators| `4`     | `1`, `4`, `7`
 NEOBENCH_VOTE|Vote for validators before the bench| empty   |`1` or empty
 
+For MacOS NEOBENCH_LOGGER should be set to `json-file` as `journald` and
+`syslog` are not supported by this architecture.
 ## Benchmark results visualisation
 
 There's a Python plotting script available for benchmark data visualisation. 

--- a/runner.sh
+++ b/runner.sh
@@ -47,7 +47,8 @@ show_help() {
 	echo "   -t                               Request timeout."
 	echo "                                    Used for RPC requests."
 	echo "                                    Example: -t 30s"
-	echo "   -l, --log                        Enable logging on consensus nodes."
+	echo "   -l, --log                        Container logging facility. Default value is none."
+	echo "                                    Example: -l journald -l syslog -l json-file"
 	echo "       --tc                         Arguments to pass to 'tc qdisc netem' inside the container."
 	echo "                                    Example: 'delay 100ms'"
 	exit 0
@@ -68,7 +69,22 @@ while test $# -gt 0; do
 
 	case $_opt in
 	-h | --help) show_help ;;
-	-l | --log) export NEOBENCH_LOGGER=journald ;;
+	-l | --log)
+		if [[ $# -gt 0 && ${1:0:1} != "-" ]]; then
+			case "$1" in
+			"syslog" | "journald" | "json-file" | "none")
+				export NEOBENCH_LOGGER="$1"
+				shift
+				;;
+			*)
+				fatal "unknown logger specified: $1"
+				;;
+			esac
+		else
+		  export NEOBENCH_LOGGER="none"
+		fi
+		;;
+
 	--vote) export NEOBENCH_VOTE=1 ;;
 
 	-v | --validators)

